### DISCUSSION TITLE
Installer build for Linux

### DISCUSSION THREAD

**Discussion by CanYing0913** [marcelotduarte/cx_Freeze#1877] [Q&A] [answered]
I want to achieve the same effect as I build .dmg and .msi installer on Linux. What is the specific Linux command to run my setup.py to do so? Appreciate for any help.

**Comment by marcelotduarte** [comment, marked as answer]
cx_Freeze has [bdist_rpm](https://cx-freeze.readthedocs.io/en/latest/setup_script.html#bdist-rpm).
Using Ubuntu/Debian you can need to install the rpm package:
`sudo apt install rpm`
I hope that you can convert the rpm to deb or rpm to tgz using the alien package:
`sudo apt install alien`

In a future I plan to implement support to appimage or snap.

**Reply by marcelotduarte** [reply]
@CanYing0913  Support for AppImage: bdist_appimage available #2054

**Reply by ntindle** [reply]
Seems like github actions don't support installing alien. Will look into more later

**Reply by marcelotduarte** [reply]
@ntindle
`sudo apt-get install -y alien fakeroot rpm`
See: https://github.com/marcelotduarte/cx_Freeze/actions/runs/9378760979/job/25822490347#step:4:1

**Reply by ntindle** [reply]
Weird. It keeps failing in act (action tester). I'll give this exact a shot in a bit. 

Edit: worked in gha but not act

**Reply by marcelotduarte** [reply]
A test: https://github.com/marcelotduarte/cx_Freeze/blob/main/tests/test_command_bdist_appimage.py#L113
The result: https://github.com/marcelotduarte/cx_Freeze/actions/runs/9378760979/job/25822490347#step:6:109

**Reply by marcelotduarte** [reply]
@ntindle Maybe you are experiencing a bug in cpio #2402 
https://github.com/marcelotduarte/cx_Freeze/pull/2402/files#diff-9859a432e878bc969b09eaa7d835320558bd8a69e8e6d34c413a9f67667663ceR111

**Reply by ntindle** [reply]
Will take a look
